docs/source/xhighfive.rst

```diff
@@ -0,0 +1,148 @@
+
+        A = xt::load_hdf5<xt::xtensor<double,2>>("example.h5", "/path/to/data");
+
+        std::cout << A << std::endl;
+
+        return 0;
+    }
+
+Advanced interface
+------------------
+
+The advanced interface provides simple free-functions that write to or read from an open ``HighFive::File``. It consists of the following functions:
+
+*   ``xt::dump(file, path, data[, xt::dump_mode])``
+
+    Dump data (e.g. a matrix) to a DataSet in a HDF5 file. The data can be: ``scalar`` (incl. ``std::string``), ``std::vector<scalar>`` (incl. ``std::string``), ``xt::xarray<scalar>``, and xt::xtensor<scalar,dim>``.
+
+*   ``xt::dump(file, path, data, i[, j])``
+
+    Dump scalar to the ``i``th item of a DataSet that is a one-dimensional array. If the DataSet does not yet exist, an extendible DataSet is created. Likewise for a two-dimensional array. The same overloads as for dump are available.
+
+*   ``data = xt::load<...>(file, path)``
+
+    Read data (e.g. a matrix) from a DataSet in a HDF5 file. The same overloads as for dump are available. The same overloads as for dump are available.
+
+*   ``data = xt::load<...>(file, path, i[, j])``
+
+    Read scalar as the ``i``th item of a DataSet that is a one-dimensional array. Likewise for a two-dimensional array. The same overloads as for dump are available.
+
+For example:
+
+.. code-block:: cpp
+
+    #include <xtensor/xio.hpp>
+    #include <xtensor-io/xhighfive.hpp>
+
+    int main()
+    {
+        HighFive::File file("example.h5", HighFive::File::Overwrite);
+
+        xt::xtensor<double,2> A = xt::ones<double>({10,5});
+
+        xt::dump(file, "/path/to/data", A);
+
+        xt::dump(file, "/path/to/data", A, xt::dump_mode::overwrite);
+
+        A = xt::load<xt::xtensor<double,2>>(file, "/path/to/data");
+
+        std::cout << A << std::endl;
+
+        return 0;
+    }
+
+Compiling & dependencies
+------------------------
+
+This library uses header only `HighFive <https://github.com/BlueBrain/HighFive>`_ library and the HDF5 library. Both should be available upon compiling and linking respectively.
+
+Manual
+^^^^^^
+
+Compiling can then proceed through
+
+.. code-block:: bash
+
+    g++ -std=c++14 -lhdf5 main.cpp
+
+    # manually set paths
+    g++ -I... -L... -std=c++14 -lhdf5 main.cpp
+
+Alternatively, HDF5 provides a wrapper command that sets the paths to the HDF5 library (not to HighFive):
+
+.. code-block:: bash
+
+    h5c++ -std=c++14 main.cpp
+
+Using CMake
+^^^^^^^^^^^
+
+The following basic structure of ``CMakeLists.txt`` can be used:
+
+.. code-block:: cmake
+
+  cmake_minimum_required(VERSION 2.8.12)
+
+  # define a project name
+  project(example)
+
+  # define empty list of libraries to link
+  set(PROJECT_LIBS "")
+
+  # enforce the C++ standard
+  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+  # set optimization level
+  set(CMAKE_BUILD_TYPE Release)
+
+  # ...
+
+  # find HDF5
+  find_package(HDF5 COMPONENTS CXX REQUIRED)
```

Comment:
I think there should also be a `find_package(HighFive REQUIRED)`

Reply:
Will do.